### DISCUSSION TITLE
ESP32 and Arduino

### DISCUSSION THREAD

**Discussion by LavredisG** [OpenEVSE/open_evse#22] [Q&A] [answered]
Hello,
I have a question regarding the firmware. If I want to simulate the openevse kit, should I download
the openevse firmware on an arduino and the WiFi gateway firmware on an ESP32 and later connect
the arduino to the esp32, or download both on the ESP32. What is the best approach on the matter?

Thank you in advance!

**Comment by chris1howell** [comment, marked as answer]
Simulating on generic development boards is not very useful and problematic because the required safety and pilot circuits are not present. All our internal development and community developers use actual OpenEVSE Hardware. 

A OpenEVSE controller, OpenEVSE WiFi board, relay, GFCI coil an EV simulator (resistors and Diode) are the bare minimum.
